numpy/_core/src/multiarray/conversion_utils.c

```diff
@@ -941,6 +941,10 @@ static int casting_parser(char const *str, Py_ssize_t length, void *data)
             *casting = NPY_SAME_KIND_CASTING;
             return 0;
         }
+        if (can_use_same_value && length == 10 && strcmp(str, "same_value") == 0) {
+            *casting = NPY_SAME_VALUE_CASTING;
```

Comment:
So, if one goes with adding explicitly that same_value casting is implied by no and equiv, I guess the question is whether in this `switch`, one could/should set `*casting = NPY_NO_CASTING | NPY_SAME_VALUE_CASTING`, or should just define `NPY_NO_CASTING` and `NPY_EQUIV_CASTING` to include it.

Reply:
> should just define NPY_NO_CASTING and NPY_EQUIV_CASTING to include it.

Yeah maybe... we would have to be very careful with old code passing the version without the flag, though.  Because this is ABI in the end, not just API.